homeassistant/helpers/discovery.py

```diff
@@ -158,18 +159,22 @@ async def async_load_platform(
 
     setup_success = True
 
-    if component not in hass.config.components:
-        setup_success = await setup.async_setup_component(hass, component, hass_config)
+    # Ensure that the underlying platform domain is setup
+    platform_domain = platform.value if isinstance(platform, Platform) else platform
```

Comment:
We could add a `frame.report` here if we want to encourage people to use the Platform enum.
For example "if platform is a string and platform could be in Platform then report the frame"

Reply:
but since Platform is a helper and not a definitive list maybe it is not necessary to report it

Reply:
I think the better long term option would be to at some point update the type again and only allow `Platform` not `str` anymore. For now it's perfectly fine to pass a `str`. As I mentioned, they behave the same.